### DISCUSSION TITLE
Change default Reclaim Policy of Vsphere-Csi-Controller

### DISCUSSION THREAD

**Discussion by Sovenique** [rancher/rke2#4105] [Q&A]
Greetings,

I'm looking for a way to change the default reclaim policy of the PVCs spawned by the plugin `rancher-vsphere-csi-101.0.0+up2.5.1-rancher1`. I tried to tweak the Deployment object of the vsphere-csi-controller so the vsphere-csi-sc StorageClass is reclaimPolicy: Delete. I can't seem to find a parameter to change it beforehand so it's always e.g. reclaimPolicy: Retain.

I can't edit the current storageClass object either, if I specify anything else than Delete is rejects it as invalid:

```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: vsphere-csi-sc
  uid: 50c79d4c-f569-4274-93c2-2f6464a96388
  resourceVersion: '423208'
  creationTimestamp: '2023-03-28T10:00:18Z'
  labels:
    app.kubernetes.io/managed-by: Helm
  annotations:
    meta.helm.sh/release-name: vsphere-csi
    meta.helm.sh/release-namespace: kube-system
    storageclass.kubernetes.io/is-default-class: 'true'
  managedFields:
    - manager: helm
      operation: Update
      apiVersion: storage.k8s.io/v1
      time: '2023-03-28T10:00:18Z'
      fieldsType: FieldsV1
      fieldsV1:
        f:allowVolumeExpansion: {}
        f:metadata:
          f:annotations:
            .: {}
            f:meta.helm.sh/release-name: {}
            f:meta.helm.sh/release-namespace: {}
            f:storageclass.kubernetes.io/is-default-class: {}
          f:labels:
            .: {}
            f:app.kubernetes.io/managed-by: {}
        f:provisioner: {}
        f:reclaimPolicy: {}
        f:volumeBindingMode: {}
  selfLink: /apis/storage.k8s.io/v1/storageclasses/vsphere-csi-sc
allowVolumeExpansion: true
allowedTopologies: []
mountOptions: []
parameters: {}
provisioner: csi.vsphere.vmware.com
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

My apologies if I've missed something in the documentation or if it's not the right place. So is this feasible at all?

Thank you so much for your time.

**Comment by brandond** [comment]
> `rancher-vsphere-csi-101.0.0+up2.5.1-rancher1`

It sounds like you've deployed this via Rancher, as this is not a version of the chart bundled by RKE2. If you're using Rancher to deploy the vsphere CSI chart, you should customize the helm chart via the Rancher UI.
